### ISSUE TITLE
Switch Advanced Search date inputs to type="date"

### ISSUE THREAD

**Issue by techgique** [open-oni/open-oni#375]
This changes the value submitted to the form to `yyyy-mm-dd` from currently specified `dd/mm/yyyy`. The helper text would have to be updated and can be handled on the back end in `_solrize_date` here: https://github.com/open-oni/open-oni/blob/fc0270e045565954ce7ed08b30a9a7221f8a1079/core/solr_index.py#L680

**Comment by jduss4** [comment]
I might have gotten a little carried away about refactoring solrize_date, but I didn't like how they were constructing the date at the end with unnecessary math.  And then I added in some range "*" type assumptions, as well.  The commit messages has more info about some of my thoughts on this, but I believe it to be functional and tests related to it are passing:

https://github.com/open-oni/open-oni/tree/date_form

I did find out in the meantime that a LOT of other tests are failing, so something to keep in mind while looking at this branch:  https://github.com/open-oni/open-oni/issues/386

**Comment by techgique** [comment]
I think setting the initial value for the start date input to `{{form.fulltextStartYear}}-01-01` and for the end date to `{{form.fulltextEndYear}}-12-31` would make it easier to use in terms of modifying the dates from today's date.

Since this would set the start date at the first day of a year and end date at the last day of a year, it would also be akin to having the inputs start like `1850(-*)` and `1923(-*)` where only the year part of the input need be changed to include entire years within the range.

In order to avoid always sending a date as part of the Solr query, we could restore the `is_start` keyword argument to `solarize_date(date, is_start=True)` and then compare the value within the function.

(Pseudocode)
```
if is_start:
    date = '*' if date == form.fulltextStartYear-01-01
else:
    date = '*' if date == form.fulltextEndYear-12-31
```

**Comment by techgique** [comment]
Closed with #389
